core/matching/snapshot.go

```diff
@@ -130,13 +122,10 @@ func (b *OrderBook) LoadState(_ context.Context, payload *types.Payload) ([]type
 }
 
 func (b *OrderBook) addOrderToMaps(o *types.Order) {
-	b.ordersByID[o.ID] = o
+	b.add(o)
 
-	if orders, ok := b.ordersPerParty[o.Party]; !ok {
-		b.ordersPerParty[o.Party] = map[string]struct{}{
-			o.ID: {},
-		}
-	} else {
-		orders[o.ID] = struct{}{}
+	// reconstruct pegged orders state
+	if o.PeggedOrder != nil {
+		b.peggedOrders[o.ID] = struct{}{}
```

Comment:
why not add this in the add(order) function?

Reply:
Moved it into `add()`.